src/support/assertRender.ts

```diff
@@ -66,40 +61,47 @@ export default function assertRender(actual: DNode | DNode[], expected: DNode |
 	const { isHNode: localIsHNode = isHNode, isWNode: localIsWNode = isWNode, ...passedDiffOptions } = (options || {}) as AssertRenderOptions;
 	const diffOptions: DiffOptions = assign({}, defaultDiffOptions, passedDiffOptions);
 
-	function assertChildren(actual: DNode[], expected: DNode[]) {
-		if (actual.length !== expected.length) {
-			throwAssertionError(actual, expected, message);
+	function assertChildren(actual?: DNode[], expected?: DNode[]) {
+		if (actual && expected) {
+			if (actual.length !== expected.length) {
+				throwAssertionError(actual, expected, `Children's length mismatch`, message);
+			}
+			actual.forEach((actualChild, index) => {
+				assertRender(actualChild, expected[index], (options || {}) as AssertRenderOptions, `[${index}]${message || ''}`);
+			});
+		}
+		else {
+			if (actual || expected) {
+				throwAssertionError(actual, expected, actual ? 'Unxpected children' : 'Expected children', message);
+			}
 		}
-		actual.forEach((actualChild, index) => {
-			assertRender(actualChild, expected[index], message);
-		});
 	}
 
 	if (Array.isArray(actual) && Array.isArray(expected)) {
 		assertChildren(actual, expected);
 	}
 	else if (Array.isArray(actual) || Array.isArray(expected)) {
-		throwAssertionError(actual, expected, message);
+		throwAssertionError(actual, expected, Array.isArray(actual) ? `Expected "${expected === null ? 'null' : typeof expected}" but got an array` : `Expected an array but got "${actual === null ? 'null' : typeof actual}"`, message);
```

Comment:
nit: This is quite long

Reply:
I refactored all the messages, you are right it was messy and unreadable.